src/character.cpp

```diff
@@ -1061,9 +1066,38 @@ void Character::set_healthy_mod(int nhealthy_mod)
 {
     healthy_mod = nhealthy_mod;
 }
-void Character::mod_healthy_mod(int nhealthy_mod)
+void Character::mod_healthy_mod(int nhealthy_mod, int cap)
 {
+    // TODO: This really should be a full morale-like system, with per-effect caps
+    //       and durations.  This version prevents any single effect from exceeding its
+    //       intended ceiling, but multiple effects will overlap instead of adding.
+
+    // Cap indicates how far the mod is allowed to shift in this direction.
+    // It can have a different sign to the mod, e.g. for items that treat
+    // extremely low health, but can't make you healthy.
+    int low_cap, high_cap;
+    if (nhealthy_mod < 0) {
+        low_cap = cap;
+        high_cap = 200;
+    } else {
+        low_cap = -200;
+        high_cap = cap;
+    }
+
+    // If we're already out-of-bounds, we don't need to do anything.
+    if (healthy_mod <= low_cap || healthy_mod >= high_cap) {
```

Comment:
Think you meant to say "will break if `healthy_mod` is out of bounds in the opposite direction to `nhealthy_mod`".

Disregard if the clarification wasn't necessary, but my tired brain took a good 30 seconds to parse out what you meant when the comment above already talks about being out-of-bounds.